test/common-tests/request-field-validation-api-tests/email-address-field-validation-api-tests.ts

```diff
@@ -0,0 +1,100 @@
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { prepareModifiedRequest } from '@ukef-test/support/helpers/request-field-validation-helper';
+import request from 'supertest';
+
+import { withRequiredFieldValidationApiTests } from './partials/require-validation';
+import { withTypeFieldValidationApiTests } from './partials/type-validation';
+
+export interface EmailFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKey extends keyof RequestBodyItem> {
+  fieldName: RequestBodyItemKey;
+  required?: boolean;
+  validRequestBody: RequestBodyItem[] | RequestBodyItem;
+  makeRequest: ((body: unknown[]) => request.Test) | ((body: unknown) => request.Test);
+  givenAnyRequestBodyWouldSucceed: () => void;
+}
+
+export const withEmailFieldValidationApiTests = <RequestBodyItem, RequestBodyItemKey extends keyof RequestBodyItem>({
+  fieldName: fieldNameSymbol,
+  required,
+  validRequestBody,
+  makeRequest,
+  givenAnyRequestBodyWouldSucceed,
+}: EmailFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKey>): void => {
+  const valueGenerator = new RandomValueGenerator();
+  const requestIsAnArray = Array.isArray(validRequestBody);
+  const requestBodyItem = requestIsAnArray ? validRequestBody[0] : validRequestBody;
+  const fieldName = fieldNameSymbol.toString();
+  const validEmail = valueGenerator.email();
+  const typeNameForErrorMessages = 'an email';
+  required = required ?? true;
+
+  describe(`${fieldName} validation`, () => {
+    beforeEach(() => {
+      givenAnyRequestBodyWouldSucceed();
+    });
+
+    it(`returns a 2xx response validRequestBody is provided`, async () => {
+      const preparedRequest = prepareModifiedRequest(requestIsAnArray, requestBodyItem);
+
+      const { status } = await makeRequest(preparedRequest);
+      expect(status).toBeGreaterThanOrEqual(200);
+      expect(status).toBeLessThan(300);
+    });
+
+    describe(`${fieldName} email specific validation`, () => {
+      it(`returns a 2xx response if ${fieldName} is correct email is "${validEmail}"`, async () => {
+        const requestWithInvalidValue = { ...requestBodyItem, [fieldNameSymbol]: validEmail };
+        const preparedRequest = prepareModifiedRequest(requestIsAnArray, requestWithInvalidValue);
+
+        const { status } = await makeRequest(preparedRequest);
+
+        expect(status).toBeGreaterThanOrEqual(200);
+        expect(status).toBeLessThan(300);
+      });
+
+      it.each([
+        {
+          invalidEmail: 'test@example',
+          expectedError: 'sendToEmailAddress must be an email',
+        },
+        {
+          invalidEmail: 'testexample.com',
+          expectedError: 'sendToEmailAddress must be an email',
+        },
+        {
+          invalidEmail: 'test@example.c',
```

Comment:
Can we please test this with two letter and four letter domains i.e. `.uk` , `.london` and etc.

Reply:
Added 11 various valid emails.

Reply:
Good coverage.